# Saloni Patel (A4-4.4)

# Importing Libraries and Data

### Importing Libraries

In [1]:
# importing libraries

import pandas as pd
import numpy as np
import os

### Importing Order Data

In [2]:
# Importing order data

df_ord = pd.read_csv(r'C:\Users\psalo\OneDrive\Study\CF - DA Immersion\A4 - Python Fundamentals for Data Analyst\A4 - Instacart Basket Analysis 07-2025\02 Data\Original Data\4.3_orders_products\orders.csv', index_col = False)


### Importing Product Data

In [3]:
#Importing Product data

df_prod = pd.read_csv(r'C:\Users\psalo\OneDrive\Study\CF - DA Immersion\A4 - Python Fundamentals for Data Analyst\A4 - Instacart Basket Analysis 07-2025\02 Data\Original Data\4.3_orders_products\products.csv', index_col = False)


### Importing Department Data

In [4]:
#Importing Department data

df_dept = pd.read_csv(r'C:\Users\psalo\OneDrive\Study\CF - DA Immersion\A4 - Python Fundamentals for Data Analyst\A4 - Instacart Basket Analysis 07-2025\02 Data\Original Data\4.4_departments\departments.csv', index_col = False)


### Checking the data import using head()

In [6]:
df_ord.head(5)

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

In [7]:
df_prod.head(5)

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [8]:
df_dept.head()

department_id       1      2       3        4        5              6  \
0    department  frozen  other  bakery  produce  alcohol  international   

           7     8                9  ...            12      13         14  \
0  beverages  pets  dry goods pasta  ...  meat seafood  pantry  breakfast   

             15          16         17      18      19    20       21  
0  canned goods  dairy eggs  household  babies  snacks  deli  missing  

[1 rows x 22 columns]

# Data Wrangling

### Dropping the 'eval_set' column from Orders df

In [5]:
df_ord.drop(columns = ['eval_set'])

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0         2539329        1             1          2                  8   
1         2398795        1             2          3                  7   
2          473747        1             3          3                 12   
3         2254736        1             4          4                  7   
4          431534        1             5          4                 15   
...           ...      ...           ...        ...                ...   
3421078   2266710   206209            10          5                 18   
3421079   1854736   206209            11          4                 10   
3421080    626363   206209            12          1                 12   
3421081   2977660   206209            13          1                 12   
3421082    272231   206209            14          6                 14   

         days_since_prior_order  
0                           NaN  
1                          15.0  
2                          21.0  
3                          29.0  
4                          28.0  
...                         ...  
3421078                    29.0  
3421079                    30.0  
3421080                    18.0  
3421081                     7.0  
3421082                    30.0  

[3421083 rows x 6 columns]

### Checking if eval_set is dropped successfully or not

In [6]:
df_ord.head(5)

order_id  user_id eval_set  order_number  order_dow  order_hour_of_day  \
0   2539329        1    prior             1          2                  8   
1   2398795        1    prior             2          3                  7   
2    473747        1    prior             3          3                 12   
3   2254736        1    prior             4          4                  7   
4    431534        1    prior             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

### Checking for Frequency of a particular data values

In [11]:
df_ord.days_since_prior_order.value_counts(dropna = False)

days_since_prior_order
30.0    369323
7.0     320608
6.0     240013
4.0     221696
3.0     217005
5.0     214503
NaN     206209
2.0     193206
8.0     181717
1.0     145247
9.0     118188
14.0    100230
10.0     95186
13.0     83214
11.0     80970
12.0     76146
0.0      67755
15.0     66579
16.0     46941
21.0     45470
17.0     39245
20.0     38527
18.0     35881
19.0     34384
22.0     32012
28.0     26777
23.0     23885
27.0     22013
24.0     20712
25.0     19234
29.0     19191
26.0     19016
Name: count, dtype: int64

### Renaming DOW in Order DF

In [12]:
df_ord.rename(columns = {'order_dow' : 'orders_day_of_week'}, inplace = True)

### Checking to see if the name change took place successfully

In [13]:
df_ord.head(3)

order_id  user_id eval_set  order_number  orders_day_of_week  \
0   2539329        1    prior             1                   2   
1   2398795        1    prior             2                   3   
2    473747        1    prior             3                   3   

   order_hour_of_day  days_since_prior_order  
0                  8                     NaN  
1                  7                    15.0  
2                 12                    21.0

### Changing Variable's Data Type

In [14]:
df_ord.describe()

order_id       user_id  order_number  orders_day_of_week  \
count  3.421083e+06  3.421083e+06  3.421083e+06        3.421083e+06   
mean   1.710542e+06  1.029782e+05  1.715486e+01        2.776219e+00   
std    9.875817e+05  5.953372e+04  1.773316e+01        2.046829e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00        0.000000e+00   
25%    8.552715e+05  5.139400e+04  5.000000e+00        1.000000e+00   
50%    1.710542e+06  1.026890e+05  1.100000e+01        3.000000e+00   
75%    2.565812e+06  1.543850e+05  2.300000e+01        5.000000e+00   
max    3.421083e+06  2.062090e+05  1.000000e+02        6.000000e+00   

       order_hour_of_day  days_since_prior_order  
count       3.421083e+06            3.214874e+06  
mean        1.345202e+01            1.111484e+01  
std         4.226088e+00            9.206737e+00  
min         0.000000e+00            0.000000e+00  
25%         1.000000e+01            4.000000e+00  
50%         1.300000e+01            7.000000e+00  
75%         1.600000e+01            1.500000e+01  
max         2.300000e+01            3.000000e+01

In [15]:
df_ord.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3421083 entries, 0 to 3421082
Data columns (total 7 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   order_id                int64  
 1   user_id                 int64  
 2   eval_set                object 
 3   order_number            int64  
 4   orders_day_of_week      int64  
 5   order_hour_of_day       int64  
 6   days_since_prior_order  float64
dtypes: float64(1), int64(5), object(1)
memory usage: 182.7+ MB


### Checking single column's data type

In [16]:
df_ord['order_id'].dtype

dtype('int64')

### Changing the data type of 'order_id' to string

In [17]:
df_ord['order_id'] = df_ord['order_id'].astype('str')

In [18]:
path = r'C:\Users\psalo\OneDrive\Study\CF - DA Immersion\A4 - Python Fundamentals for Data Analyst\A4 - Instacart Basket Analysis 07-2025'

### Checking the data type of 'order id' after the change

In [19]:
df_ord['order_id'].dtype

dtype('O')

In [55]:
df_ord.info

<bound method DataFrame.info of         order_id user_id eval_set  order_number  orders_day_of_week  \
0        2539329       1    prior             1                   2   
1        2398795       1    prior             2                   3   
2         473747       1    prior             3                   3   
3        2254736       1    prior             4                   4   
4         431534       1    prior             5                   4   
...          ...     ...      ...           ...                 ...   
3421078  2266710  206209    prior            10                   5   
3421079  1854736  206209    prior            11                   4   
3421080   626363  206209    prior            12                   1   
3421081  2977660  206209    prior            13                   1   
3421082   272231  206209    train            14                   6   

         order_hour_of_day  days_since_last_order  
0                        8                    NaN  
1          

### Changing the rows to columns (transpose) for Department DF

In [20]:
df_dept.T

0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

### Creating a new df with the transpose data

In [21]:
df_dept2 = df_dept.T

In [22]:
df_dept2.head

<bound method NDFrame.head of                              0
department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing>

### Inserting index so headers are properly displayed

In [23]:
df_dept2.reset_index()

index                0
0   department_id       department
1               1           frozen
2               2            other
3               3           bakery
4               4          produce
5               5          alcohol
6               6    international
7               7        beverages
8               8             pets
9               9  dry goods pasta
10             10             bulk
11             11    personal care
12             12     meat seafood
13             13           pantry
14             14        breakfast
15             15     canned goods
16             16       dairy eggs
17             17        household
18             18           babies
19             19           snacks
20             20             deli
21             21          missing

### Creating new header

In [24]:
# Saving a header which is line 0 to a new variable called new_header

new_header = df_dept2.iloc[0]

In [25]:
# see what a new_header looks like

new_header

0    department
Name: department_id, dtype: object

In [26]:
# creating another version of dept2. It is called deptnew. it has the all the data from dept2 starting from row 1.

df_deptnew = df_dept2[1:]

In [27]:
# see what a new deptnew looks like

df_deptnew

0
1            frozen
2             other
3            bakery
4           produce
5           alcohol
6     international
7         beverages
8              pets
9   dry goods pasta
10             bulk
11    personal care
12     meat seafood
13           pantry
14        breakfast
15     canned goods
16       dairy eggs
17        household
18           babies
19           snacks
20             deli
21          missing

In [28]:
#adding a header to deptnew by putting a new_header variable as a columns name for the new table

df_deptnew.columns = new_header


In [29]:
# See what the df_deptnew looks like 

df_deptnew

department_id       department
1                       frozen
2                        other
3                       bakery
4                      produce
5                      alcohol
6                international
7                    beverages
8                         pets
9              dry goods pasta
10                        bulk
11               personal care
12                meat seafood
13                      pantry
14                   breakfast
15                canned goods
16                  dairy eggs
17                   household
18                      babies
19                      snacks
20                        deli
21                     missing

# Creating a new data dictionary

In [30]:
# turning df_deptnew into a data dictionary

dept_dict = df_deptnew.to_dict('index')

In [31]:
# see how newly created dept_dict looks like

dept_dict

{'1': {'department': 'frozen'},
 '2': {'department': 'other'},
 '3': {'department': 'bakery'},
 '4': {'department': 'produce'},
 '5': {'department': 'alcohol'},
 '6': {'department': 'international'},
 '7': {'department': 'beverages'},
 '8': {'department': 'pets'},
 '9': {'department': 'dry goods pasta'},
 '10': {'department': 'bulk'},
 '11': {'department': 'personal care'},
 '12': {'department': 'meat seafood'},
 '13': {'department': 'pantry'},
 '14': {'department': 'breakfast'},
 '15': {'department': 'canned goods'},
 '16': {'department': 'dairy eggs'},
 '17': {'department': 'household'},
 '18': {'department': 'babies'},
 '19': {'department': 'snacks'},
 '20': {'department': 'deli'},
 '21': {'department': 'missing'}}

In [32]:
# recalling product data to see how it can relate to dept data

df_prod.head(5)

product_id                                       product_name  aisle_id  \
0           1                         Chocolate Sandwich Cookies        61   
1           2                                   All-Seasons Salt       104   
2           3               Robust Golden Unsweetened Oolong Tea        94   
3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4           5                          Green Chile Anytime Sauce         5   

   department_id  prices  
0             19     5.8  
1             13     9.3  
2              7     4.5  
3              1    10.5  
4             13     4.3

In [33]:
# checking the department of the product_id = 1

print(dept_dict.get('19'))

{'department': 'snacks'}


### Creating a Subset of snacks category

In [34]:
df_snacks =  df_prod[df_prod['department_id']==19]

In [35]:
# see what the subset df_snacks look like

df_snacks

product_id                                   product_name  aisle_id  \
0               1                     Chocolate Sandwich Cookies        61   
15             16                  Mint Chocolate Flavored Syrup       103   
24             25        Salted Caramel Lean Protein & Fiber Bar         3   
31             32                  Nacho Cheese White Bean Chips       107   
40             41    Organic Sourdough Einkorn Crackers Rosemary        78   
...           ...                                            ...       ...   
49666       49662                   Bacon Cheddar Pretzel Pieces       107   
49669       49665  Super Dark Coconut Ash & Banana Chocolate Bar        45   
49670       49666                  Ginger Snaps Snacking Cookies        61   
49675       49671                           Milk Chocolate Drops        45   
49683       49679                        Famous Chocolate Wafers        61   

       department_id  prices  
0                 19     5.8  
15                19     5.2  
24                19     1.9  
31                19     4.9  
40                19     6.5  
...              ...     ...  
49666             19     3.6  
49669             19     6.9  
49670             19     5.2  
49675             19     3.0  
49683             19     6.0  

[6264 rows x 5 columns]

In [36]:
# Retriving the same information as subset df_snacks but directly from the df_prod

df_prod[df_prod['department_id']==19] 

product_id                                   product_name  aisle_id  \
0               1                     Chocolate Sandwich Cookies        61   
15             16                  Mint Chocolate Flavored Syrup       103   
24             25        Salted Caramel Lean Protein & Fiber Bar         3   
31             32                  Nacho Cheese White Bean Chips       107   
40             41    Organic Sourdough Einkorn Crackers Rosemary        78   
...           ...                                            ...       ...   
49666       49662                   Bacon Cheddar Pretzel Pieces       107   
49669       49665  Super Dark Coconut Ash & Banana Chocolate Bar        45   
49670       49666                  Ginger Snaps Snacking Cookies        61   
49675       49671                           Milk Chocolate Drops        45   
49683       49679                        Famous Chocolate Wafers        61   

       department_id  prices  
0                 19     5.8  
15                19     5.2  
24                19     1.9  
31                19     4.9  
40                19     6.5  
...              ...     ...  
49666             19     3.6  
49669             19     6.9  
49670             19     5.2  
49675             19     3.0  
49683             19     6.0  

[6264 rows x 5 columns]

In [37]:
# another way of looking up the same information using loc

df_snacks2 = df_prod.loc[df_prod['department_id'].isin([19])]

In [38]:
df_snacks2

product_id                                   product_name  aisle_id  \
0               1                     Chocolate Sandwich Cookies        61   
15             16                  Mint Chocolate Flavored Syrup       103   
24             25        Salted Caramel Lean Protein & Fiber Bar         3   
31             32                  Nacho Cheese White Bean Chips       107   
40             41    Organic Sourdough Einkorn Crackers Rosemary        78   
...           ...                                            ...       ...   
49666       49662                   Bacon Cheddar Pretzel Pieces       107   
49669       49665  Super Dark Coconut Ash & Banana Chocolate Bar        45   
49670       49666                  Ginger Snaps Snacking Cookies        61   
49675       49671                           Milk Chocolate Drops        45   
49683       49679                        Famous Chocolate Wafers        61   

       department_id  prices  
0                 19     5.8  
15                19     5.2  
24                19     1.9  
31                19     4.9  
40                19     6.5  
...              ...     ...  
49666             19     3.6  
49669             19     6.9  
49670             19     5.2  
49675             19     3.0  
49683             19     6.0  

[6264 rows x 5 columns]

In [39]:
# defining the path

path = r'C:\Users\psalo\OneDrive\Study\CF - DA Immersion\A4 - Python Fundamentals for Data Analyst\A4 - Instacart Basket Analysis 07-2025'

# Assignment A4 - 4.4

#### 2 - Find another identifier variable in the df_ords dataframe that doesn’t need to be included in your analysis as a numeric variable and change it to a suitable format.

user_id is the variable that can use the data type change.



In [40]:
# finding out the data type for the variable

df_ord['user_id'].dtype

dtype('int64')

In [41]:
# changing the data type for the variable from int64 to str

df_ord['user_id'] = df_ord['user_id'].astype('str')

In [42]:
# checking the data type now

df_ord['user_id'].dtype


dtype('O')

#### 3 - Look for a variable in your df_ords dataframe with an unintuitive name and change its name without overwriting the dataframe.

variable name 'days_since_prior_order'  will be name changed to days_since_last_order

In [43]:
# changing the variable name from 'days_since_prior_order' to 'days_since_last_order'

df_ord.rename(columns = {'days_since_prior_order' : 'days_since_last_order'}, inplace = True)

In [56]:
# checking the varibale name change

df_ord.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3421083 entries, 0 to 3421082
Data columns (total 7 columns):
 #   Column                 Dtype  
---  ------                 -----  
 0   order_id               object 
 1   user_id                object 
 2   eval_set               object 
 3   order_number           int64  
 4   orders_day_of_week     int64  
 5   order_hour_of_day      int64  
 6   days_since_last_order  float64
dtypes: float64(1), int64(3), object(3)
memory usage: 182.7+ MB


#### 4 - Your client wants to know what the busiest hour is for placing orders. Find the frequency of the corresponding variable and share your findings.

10 am is the most fequent time of the day when customers have placed orders. There were 288,418 orders in that hour. Closely following is 11am with 284,728 orders.

In [45]:
# checking the fequency of the values in column order_hour_of_day 

df_ord.order_hour_of_day.value_counts(dropna = False)

order_hour_of_day
10    288418
11    284728
15    283639
14    283042
13    277999
12    272841
16    272553
9     257812
17    228795
18    182912
8     178201
19    140569
20    104292
7      91868
21     78109
22     61468
23     40043
6      30529
0      22758
1      12398
5       9569
2       7539
4       5527
3       5474
Name: count, dtype: int64

#### 5 - Determine the meaning behind a value of 4 in the "department_id" column within the df_prods dataframe using a data dictionary.

Departmetn_id 4 = 'produce'

In [46]:
print(dept_dict.get('4'))

{'department': 'produce'}


#### 6 - The sales team in your client’s organization wants to know more about breakfast item sales. Create a subset containing only the required information.

The subset created for breakfast item from the df_prod is called 'df_breakfast'. There are 1,116 items in this category. 

In [47]:
# breakfast category is department_id = 14 so we will look up the information only for that value from df_prod 

df_breakfast =  df_prod[df_prod['department_id']==14]

In [48]:
df_breakfast

product_id                                       product_name  \
27             28                                  Wheat Chex Cereal   
33             34                                                NaN   
67             68                            Pancake Mix, Buttermilk   
89             90                                       Smorz Cereal   
210           211   Gluten Free Organic Cereal Coconut Maple Vanilla   
...           ...                                                ...   
49330       49326                            Cereal Variety Fun Pack   
49395       49391            Light and Fluffy Buttermilk Pancake Mix   
49547       49543                          Chocolate Cheerios Cereal   
49637       49633               Shake 'N Pour Buttermilk Pancake Mix   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   

       aisle_id  department_id  prices  
27          121             14    10.1  
33          121             14    12.2  
67          130             14    13.7  
89          121             14     3.9  
210         130             14     3.6  
...         ...            ...     ...  
49330       121             14     9.1  
49395       130             14     2.0  
49547       121             14    10.8  
49637       130             14    14.2  
49667        57             14    10.2  

[1116 rows x 5 columns]

#### 7 - They’d also like to see details about products that customers might use to throw dinner parties. Your task is to find all observations from the entire dataframe that include items from the following departments: alcohol, deli, beverages, and meat/seafood. You’ll need to present this subset to your client.

new subset 'df_partysupply' has been created. It includes the data for various departments. (1=frozen, 3=bakery, 4=product, 5=alcohol, 7=beverages, 12=meat seafood, 19=snacks, and 20=deli. We have 21,121 rows of data in this subset.

In [49]:
# Lets checck the departmetn dictionary so we can determine which departments should be look up in our inquiery 

dept_dict

{'1': {'department': 'frozen'},
 '2': {'department': 'other'},
 '3': {'department': 'bakery'},
 '4': {'department': 'produce'},
 '5': {'department': 'alcohol'},
 '6': {'department': 'international'},
 '7': {'department': 'beverages'},
 '8': {'department': 'pets'},
 '9': {'department': 'dry goods pasta'},
 '10': {'department': 'bulk'},
 '11': {'department': 'personal care'},
 '12': {'department': 'meat seafood'},
 '13': {'department': 'pantry'},
 '14': {'department': 'breakfast'},
 '15': {'department': 'canned goods'},
 '16': {'department': 'dairy eggs'},
 '17': {'department': 'household'},
 '18': {'department': 'babies'},
 '19': {'department': 'snacks'},
 '20': {'department': 'deli'},
 '21': {'department': 'missing'}}

In [50]:
# We are retriving the information from varous categories as party supply can be overlapping in this categories. 

df_partysupply = df_prod.loc[df_prod['department_id'].isin([1,3,4,5,7,12,19,20])]

In [51]:
# lets see what df_partysupply looks like

df_partysupply

product_id                                       product_name  \
0               1                         Chocolate Sandwich Cookies   
2               3               Robust Golden Unsweetened Oolong Tea   
3               4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
6               7                     Pure Coconut Water With Orange   
7               8                  Cut Russet Potatoes Steam N' Mash   
...           ...                                                ...   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   

       aisle_id  department_id  prices  
0            61             19     5.8  
2            94              7     4.5  
3            38              1    10.5  
6            98              7     4.4  
7           116              1     1.1  
...         ...            ...     ...  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  

[21121 rows x 5 columns]

#### 8 - It’s important that you keep track of total counts in your dataframes. How many rows does the last dataframe you created have?

new subset 'df_partysupply' has been created. It includes the data for various departments. (1=frozen, 3=bakery, 4=product, 5=alcohol, 7=beverages, 12=meat seafood, 19=snacks, and 20=deli. We have 21,121 rows of data in this subset.

#### 9 - Someone from the data engineers team in Instacart thinks they’ve spotted something strange about the customer with a "user_id" of “1.” Extract all the information you can about this user.

There are 11 orders for the user with 'user_id' aa 1

In [52]:
# We are retriving the information for 'user_id' = 1

df_ord[df_ord['user_id']=='1'] 

order_id user_id eval_set  order_number  orders_day_of_week  \
0   2539329       1    prior             1                   2   
1   2398795       1    prior             2                   3   
2    473747       1    prior             3                   3   
3   2254736       1    prior             4                   4   
4    431534       1    prior             5                   4   
5   3367565       1    prior             6                   2   
6    550135       1    prior             7                   1   
7   3108588       1    prior             8                   1   
8   2295261       1    prior             9                   1   
9   2550362       1    prior            10                   4   
10  1187899       1    train            11                   4   

    order_hour_of_day  days_since_last_order  
0                   8                    NaN  
1                   7                   15.0  
2                  12                   21.0  
3                   7                   29.0  
4                  15                   28.0  
5                   7                   19.0  
6                   9                   20.0  
7                  14                   14.0  
8                  16                    0.0  
9                   8                   30.0  
10                  8                   14.0

#### 10 - You also need to provide some details about this user’s behavior. What basic stats can you provide based on the information you have?

Customer in question is identified by user_id = 1.

1. There are 11 orders for this user. 
2. 7 out of 11 orders are placed between 7am to 9am. 
3. 7 out of 11 orders are placed early in the week. 
4. There are no orders placed later in the week.
5. Order 2295261 associated with this user was placed on the same day or very next day of the previous order.
6. This user is not a weekly shopper as 7 out of 11 orders have been placed after 2 weeks or more of the previous orders. 

  

#### 12 - Export your df_ords dataframe as “orders_wrangled.csv” in your “Prepared Data” folder.

In [57]:
num_rows = len(df_ord)  
print(f"{num_rows} rows exported to CSV.")

df_ord.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_wrangled.csv'))

3421083 rows exported to CSV.


#### 13 - Export the df_dep_t_new dataframe as “departments_wrangled.csv” in your “Prepared Data” folder so that you have a “.csv” file of your departments data in the correct format.

In [59]:
num_rows = len(df_deptnew)  
print(f"{num_rows} rows exported to CSV.")

df_deptnew.to_csv(os.path.join(path, '02 Data','Prepared Data', 'departments_wrangled.csv'))

21 rows exported to CSV.
